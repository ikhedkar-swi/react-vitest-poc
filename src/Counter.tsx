import { useState } from "react";

export function Counter() {
  const [count, setCount] = useState(0);

  return (
    <div>
      <p id="labelCounter">You clicked {count} times</p>
      <button id="btnClick" onClick={() => setCount(count + 1)}>
        Click me!
      </button>
    </div>
  );
}
