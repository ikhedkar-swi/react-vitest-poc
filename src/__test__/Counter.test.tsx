import { describe, expect, test } from "vitest";
import { create } from "react-test-renderer";
import { Counter } from "../Counter";

describe("Counter", () => {
  test("should show click count as 0 on initial render", () => {
    const counter = create(<Counter />);
    expect(counter).toMatchSnapshot();
  });

  test("should show the incremented counter on button clicks", async () => {
    const counter = create(<Counter />);
    const instance = counter.root;
    const button = await instance.findByType("button");
    const counterTextElem = await instance.findByProps({ id: "labelCounter" });

    expect(counter.toJSON()).toMatchSnapshot();
    expect(counterTextElem.props.children).toEqual([
      "You clicked ",
      0,
      " times",
    ]);
    expect(button.props.children).toBe("Click me!");

    button.props.onClick();
    expect(counter.toJSON()).toMatchSnapshot();
    expect(counterTextElem.props.children).toEqual([
      "You clicked ",
      1,
      " times",
    ]);

    button.props.onClick();
    expect(counter.toJSON()).toMatchSnapshot();
    expect(counterTextElem.props.children).toEqual([
      "You clicked ",
      2,
      " times",
    ]);

    button.props.onClick();
    button.props.onClick();
    button.props.onClick();
    expect(counter.toJSON()).toMatchSnapshot();
    expect(counterTextElem.props.children).toEqual([
      "You clicked ",
      5,
      " times",
    ]);
  });
});
