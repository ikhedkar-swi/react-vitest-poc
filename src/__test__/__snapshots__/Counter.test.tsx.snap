// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter > should show click count as 0 on initial render 1`] = `
<div>
  <p
    id="labelCounter"
  >
    You clicked 
    0
     times
  </p>
  <button
    id="btnClick"
    onClick={[Function]}
  >
    Click me!
  </button>
</div>
`;

exports[`Counter > should show the incremented counter on button clicks 1`] = `
<div>
  <p
    id="labelCounter"
  >
    You clicked 
    0
     times
  </p>
  <button
    id="btnClick"
    onClick={[Function]}
  >
    Click me!
  </button>
</div>
`;

exports[`Counter > should show the incremented counter on button clicks 2`] = `
<div>
  <p
    id="labelCounter"
  >
    You clicked 
    1
     times
  </p>
  <button
    id="btnClick"
    onClick={[Function]}
  >
    Click me!
  </button>
</div>
`;

exports[`Counter > should show the incremented counter on button clicks 3`] = `
<div>
  <p
    id="labelCounter"
  >
    You clicked 
    2
     times
  </p>
  <button
    id="btnClick"
    onClick={[Function]}
  >
    Click me!
  </button>
</div>
`;

exports[`Counter > should show the incremented counter on button clicks 4`] = `
<div>
  <p
    id="labelCounter"
  >
    You clicked 
    5
     times
  </p>
  <button
    id="btnClick"
    onClick={[Function]}
  >
    Click me!
  </button>
</div>
`;
